release/release_test.go

```diff
@@ -149,8 +151,34 @@ func (s *ReleaseTestSuite) TestReleaseInfo(c *C) {
 }
 
 func (s *ReleaseTestSuite) TestForceDevMode(c *C) {
+
+	runTest := func(apparmorLevel apparmor.LevelType, cgroupVersion int, expect bool) {
+		restore := apparmor.MockLevel(apparmorLevel)
+		defer restore()
+		restore = cgroup.MockVersion(cgroupVersion, nil)
+		defer restore()
+		devMode := release.ReleaseInfo.ForceDevMode()
+		c.Check(devMode, Equals, expect, Commentf("unexpected force-dev-mode for AppArmor level %v cgroup v%v", apparmorLevel, cgroupVersion))
+	}
+
+	for _, tc := range []struct {
+		apparmorLevel apparmor.LevelType
+		cgroupVersion int
+		exp           bool
+	}{
+		{apparmor.Full, cgroup.V1, false},
+		{apparmor.Partial, cgroup.V1, true},
+		// unified mode
+		{apparmor.Full, cgroup.V2, true},
+		{apparmor.Partial, cgroup.V2, true},
+	} {
+		runTest(tc.apparmorLevel, tc.cgroupVersion, tc.exp)
+	}
+}
+
+func (s *ReleaseTestSuite) TestMockForceDevMode(c *C) {
 	for _, devmode := range []bool{true, false} {
-		release.MockForcedDevmode(devmode)
+		defer release.MockForcedDevmode(devmode)()
```

Comment:
Not sure what else we do in this package but I prefer the slightly more explicit "restore := MockForceDevMode(..)\ndefer restre" ?